server/engine/src/main/java/org/eclipse/lsp/cobol/cli/command/CliAnalysis.java

```diff
@@ -37,55 +46,76 @@
-            required = true,
-            description = "The COBOL program file.")
-    private File src;
-
     @CommandLine.ArgGroup(multiplicity = "1")
-    private Args args;
+    private InputConfig inputConfig = new InputConfig();
+
+    @CommandLine.ArgGroup()
+    private Args args = new Args();
 
     @CommandLine.Option(
             description = "Supported dialect values: ${COMPLETION-CANDIDATES}",
             names = {"-d", "--dialect"},
             defaultValue = "COBOL")
     private CobolLanguageId dialect;
 
+    @CommandLine.Option(
+            description = "Hide diagnostics",
+            names = {"-nd", "--no-diag", "--no-diagnostic", "--no-diagnostics"}
+    )
+    private boolean hideDiagnostics;
+
+    @CommandLine.ArgGroup(exclusive = false)
+    private ExtendedSourceConfig extendedSourceConfig = new ExtendedSourceConfig();
+
+
     @Override
     public Integer call() throws Exception {
-        if (args.workspaceConfig != null)
-            parent.initProcessorGroupsReader(args.workspaceConfig.workspace);
+        if (args.workspaceConfig != null) {
+            try {
+                parent.initProcessorGroupsReader(args.workspaceConfig.workspace);
+            } catch (Exception e) {
+                return 2;
+            }
+        }
+
+        if (inputConfig.useStdin) {
+            createTemporaryFileFromInput();
+        }
 
         Injector diCtx = Guice.createInjector(new CliModule());
         CliClientProvider cliClientProvider = diCtx.getInstance(CliClientProvider.class);
 
         cliClientProvider.setCpyPaths(createCopybooksPaths());
         cliClientProvider.setCpyExt(createCopybooksExtensions());
         JsonObject result = new JsonObject();
-        result.addProperty("uri", src.toURI().toString());
+        result.addProperty("uri", (inputConfig.useStdin) ? "N/A (User Input)" : inputConfig.src.toURI().toString());
         result.addProperty("language", dialect.getId());
         try {
-            Cli.Result analysisResult = parent.runAnalysis(src, dialect, diCtx, true);
+            Cli.Result analysisResult = parent.runAnalysis(inputConfig.src, dialect, diCtx, true);
             parent.addTiming(result, analysisResult.ctx.getBenchmarkSession());
-            JsonArray diagnostics = new JsonArray();
-            analysisResult
-                    .ctx
-                    .getAccumulatedErrors()
-                    .forEach(
-                            err -> {
-                                JsonObject diagnostic = CliUtils.diagnosticToJson(err);
-                                diagnostics.add(diagnostic);
-                            });
-            result.add("diagnostics", diagnostics);
-            result.addProperty("lines", String.valueOf(analysisResult.ctx.getExtendedDocument().toString().split("\n").length));
-            result.addProperty("size", String.valueOf(analysisResult.ctx.getExtendedDocument().toString().length()));
+            if (!hideDiagnostics) {
```

Comment:
This block can be moved to the private method to decrease the complexity

Reply:
Implemented in latest commit.